lib/diaspora_federation/entity.rb

```diff
@@ -96,12 +79,16 @@ def to_h
     def to_xml
       doc = Nokogiri::XML::DocumentFragment.new(Nokogiri::XML::Document.new)
       Nokogiri::XML::Element.new(self.class.entity_name, doc).tap do |root_element|
-        xml_elements.each do |name, value|
+        normalized_properties.each do |name, value|
```

Comment:
Use `to_h` here. And `#to_h` should use `normalized_properties` instead of `properties`.